### PR TITLE
Update go mod to 1.17

### PR DESCRIPTION
**Pull request by mat007**
Changed `1.16` to `1.17` then did a `go mod tidy`.

### CODE REVIEW

**Review by djs55 — APPROVED** [review]
LGTM!

File: go.mod [hunk 1]
```diff
@@ -1,44 +1,68 @@
...
-go 1.16
+go 1.17
 
 require (
 	github.com/Microsoft/go-winio v0.4.11
+	github.com/google/uuid v0.0.0-20170814143639-7e072fc3a7be
+	github.com/labstack/echo v3.3.10+incompatible
+	github.com/linuxkit/virtsock v0.0.0-20180830132707-8e79449dea07
+	github.com/pkg/errors v0.8.1-0.20181023235946-059132a15dd0
+	github.com/sirupsen/logrus v1.3.0
```
Comment (thaJeztah) [inline go.mod:11]: we should update this to a more current version; IIRC, this old version is known to not be compatible with go1.13 and up

File: go.mod [hunk 2]
```diff
@@ -1,44 +1,68 @@
...
 
-go 1.16
+go 1.17
 
 require (
 	github.com/Microsoft/go-winio v0.4.11
+	github.com/google/uuid v0.0.0-20170814143639-7e072fc3a7be
+	github.com/labstack/echo v3.3.10+incompatible
+	github.com/linuxkit/virtsock v0.0.0-20180830132707-8e79449dea07
+	github.com/pkg/errors v0.8.1-0.20181023235946-059132a15dd0
```
Comment (thaJeztah) [inline go.mod:10]: This version is not compatible with go1.13 error wrapping; we should update that as well

File: go.mod [hunk 3]
```diff
@@ -1,44 +1,68 @@
 module github.com/moby/vpnkit
 
-go 1.16
+go 1.17
 
 require (
 	github.com/Microsoft/go-winio v0.4.11
+	github.com/google/uuid v0.0.0-20170814143639-7e072fc3a7be
```
Comment (thaJeztah) [inline go.mod:7]: this one has tagged releases, and (I may be mistaken ) but there may be security issues with older version

**Pull request merged**